### PR TITLE
feat: Add reference doc url for error responses for git APIs

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> Earlier we missed adding the ref doc for pull conflicts and push error where we have upstream changes which needs to be pulled in local to sync local and remote.

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/GitFileUtils.java [hunk 1]
```diff
@@ -158,7 +158,7 @@
     public Mono<ApplicationJson> reconstructApplicationFromGitRepo(String organisationId,
                                                              String defaultApplicationId,
                                                              String repoName,
-                                                             String branchName) throws GitAPIException, IOException {
+                                                             String branchName) {
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/GitFileUtils.java:162]: Any specific reason on why we are removing this now ?

**Review by AnaghHegde — COMMENTED** [review]
The PR name and the change set does not match. Can you rename the PR? Most of the changes in the PR are improving the error handling messages.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/GitFileUtils.java (hunk 1 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/GitFileUtils.java:162]: This is not thrown at all so though of removing this. Earlier when we had this method as synchronous at that time this made sense

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**